command/agent/cache/api_proxy.go

```diff
@@ -76,6 +77,20 @@ func (ap *APIProxy) Send(ctx context.Context, req *SendRequest) (*SendResponse,
 	// the client doesn't manually set the header. Removing any Accept-Encoding header allows the
 	// transparent compression to occur.
 	req.Request.Header.Del("Accept-Encoding")
+
+	if req.Request.Header == nil {
+		req.Request.Header = make(map[string][]string)
```

Comment:
same as above comment

Reply:
Thanks for the suggestion! I guess I must have missed this then copied it throughout. Adding a PR with the fix.